sled-agent/src/sim/storage.rs

```diff
@@ -317,15 +365,291 @@ impl CrucibleDataInner {
             .filter(|rs| rs.state != State::Destroyed)
             .count();
 
-        info!(
-            self.log,
-            "is_empty non_destroyed_regions {} snapshots {} running_snapshots {}",
-            non_destroyed_regions,
-            snapshots,
-            running_snapshots,
-        );
+        let empty = non_destroyed_regions == 0
+            && snapshots == 0
+            && running_snapshots == 0;
+
+        if !empty {
+            info!(
+                self.log,
+                "is_empty state: {:?}",
+                serde_json::to_string(&self).unwrap(),
+            );
+
+            info!(
+                self.log,
+                "is_empty non_destroyed_regions {} snapshots {} running_snapshots {}",
+                non_destroyed_regions,
+                snapshots,
+                running_snapshots,
+            );
+        }
+
+        empty
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use omicron_test_utils::dev::test_setup_log;
+
+    /// Validate that the Crucible agent reuses ports
```

Comment:
Is this testing the sim agent, or the real agent?

Reply:
If it's all testing the sim, we should make it clear in the comments that is what we are testing, so I don't come back later and think this is testing a real agent :)

Reply:
Done! see bbca204fe